inst/programs/jax-stablehlo-update-param.mlir

```diff
@@ -0,0 +1,16 @@
+#loc1 = loc("x")
+#loc2 = loc("grad")
+module @jit_update_param attributes {jax.uses_shape_polymorphism = false, mhlo.num_partitions = 1 : i32, mhlo.num_replicas = 1 : i32} {
+  func.func public @main(%arg0: tensor<1000000xf32> {tf.aliasing_output = 0 : i32} loc("x"), %arg1: tensor<1000000xf32> loc("grad")) -> (tensor<1000000xf32> {jax.result_info = "result"}) {
```

Comment:
@sebffischer It seems that the `tf.aliasing_output` annotation here is what actually trigegrs the donation of the buffer. I'm not sure we can always set that, or if we only add it when we're sure it's possible to donate that value.